### ISSUE TITLE
Suggestion: disparity filter algorithm

### ISSUE THREAD

**Issue by briatte** [igraph/rigraph#119]
tl;dr – would like to add [this function](https://gist.github.com/briatte/ee36be88739ec09ce3ec) to `igraph`

Dear @gaborcsardi,

@alessandrobessi and myself would like to suggest adding the disparity filter algorithm by [Serrano et al. 2009](http://arxiv.org/pdf/0904.2389.pdf) to `igraph`.

The algorithm implements a simple method to identify the 'backbone' of a weighted graph (basically, using a null model instead of a static value to clip lowly weighted edges).

The algorithm has been [available in Python](https://github.com/aekpalakorn/python-backbone-network) for a while. Earlier this year, @alessandrobessi wrote a first version in R, to which I added support for directed networks, based on Serrano et al.'s appendix.

[Here's our code so far](https://gist.github.com/briatte/ee36be88739ec09ce3ec). It is, unfortunately, much slower than the Python equivalent at that stage, although we might be able to speed it up by calling `igraph`'s C functions.

If you feel that this function might fit in `igraph`, just let me know and I'll file a proper PR with the code plus coverage tests.

Best wishes,

François

**Comment by gaborcsardi** [comment]
Sorry for the long wait.

The thing is, igraph is already huge, so maybe the best would be to put this in its own package.

**Comment by alessandrobessi** [comment]
Ok! Thanks!
